packages/core/build/tokens/build.ts

```diff
@@ -0,0 +1,91 @@
+import { baseTheme } from './tokens.js';
+import { CdsTheme, Token } from './utils.js';
+import * as fs from 'fs';
+
+fs.mkdir('../../src/styles/tokens/generated', { recursive: true }, () => console.log('Building Tokens...'));
+fs.writeFileSync('../../src/styles/tokens/generated/tokens.json', JSON.stringify(flattenTheme(baseTheme), null, 2));
+fs.writeFileSync('../../src/styles/tokens/generated/_index.scss', buildSassTokens(baseTheme));
+fs.writeFileSync('../../src/styles/tokens/generated/_public.scss', buildCSSTokens(baseTheme));
+
+function buildSassTokens(theme: CdsTheme) {
+  return Object.entries(flattenTheme(theme))
+    .map(tokenEntry => `${tokenToSass(tokenEntry[1])}\n`)
+    .join('')
+    .trim();
+}
+
+function buildCSSTokens(theme: CdsTheme) {
+  const tokens = Object.entries(flattenTheme(theme));
+  return `
+:root {
+  ${tokens
+    .map(tokenEntry => `  --cds-${camelCaseToKebab(tokenEntry[1].name)}: ${convertCSSValue(tokenEntry[1])};\n`)
+    .join('')
+    .trim()}
+}
+
+[cds-base-font='16'] {
+  --cds-global-base-font-size: 100%;
+  ${tokens
+    .filter(tokenEntry => typeof tokenEntry[1].value === 'number')
+    .map(tokenEntry => `  --cds-${camelCaseToKebab(tokenEntry[1].name)}: ${convertCSSValue(tokenEntry[1], 16)};\n`)
+    .join('')
+    .trim()}
+}`;
+}
+
+function tokenToSass(token: Token) {
+  const propName = camelCaseToKebab(token.name);
+  let staticValue;
+
+  if (typeof token.value === 'number') {
+    staticValue = convertCSSValue(token);
+  } else if (typeof token.value === 'string' && token.value.slice(-2) === 'em') {
+    staticValue = token.value;
+  }
+
+  const staticVar = staticValue ? `\n$cds-${propName}-static: ${staticValue};` : '';
+  const dynamicVar = `$cds-${propName}: var(--cds-${camelCaseToKebab(token.name)}, ${convertCSSValue(token)});`;
+
+  return `${dynamicVar}${staticVar}`;
+}
+
+function flattenTheme(theme: any) {
+  function flatten(config: any, parent = ''): { [key: string]: Token } {
+    return Object.entries(config)
+      .map(([key, value]: [string, any]) => {
+        if (typeof value === 'object' && !(value instanceof Token)) {
+          return flatten(value, `${parent}${key[0].toUpperCase() + key.slice(1)}`);
+        }
+
+        value.name = `${parent[0].toLowerCase() + parent.slice(1)}${key[0].toUpperCase() + key.slice(1)}`
+          .replace('aliases', 'alias')
+          .replace('Value', '');
+        return { [value.name]: value };
+      })
+      .reduce((prev, next) => ({ ...prev, ...next }), {});
+  }
+
+  return flatten(theme);
+}
```

Comment:
This is the build script that generates the tokens. Similar to the prior but removes Theo as a dependency and simply uses TS objects instead of yml.